eng/Version.Details.xml

```diff
@@ -72,9 +72,9 @@
       <Uri>https://github.com/nuget/nuget.client</Uri>
       <Sha>707c46e558b2b027d7ae942028c369e26545f10a</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="4.9.0-3.23602.1">
+    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="4.9.0-2.23575.1">
```

Comment:
May I clarify if this is something required for or in scope of this PR? 
Usually those updates are done automatically for example: https://github.com/dotnet/msbuild/pull/9517/files

Reply:
Thank you @jrdodds for clarification! 
I just checked this file the main branch and the vs17.9 branch
main: https://github.com/dotnet/msbuild/blob/main/eng/Version.Details.xml
vs17.9: https://github.com/dotnet/msbuild/blob/vs17.9/eng/Version.Details.xml
And it seems that the version in both of the branches is "4.9.0-3.23602.1", maybe the main branch was not up to date when it was merged. I can see that the version got updated last week in this PR: https://github.com/dotnet/msbuild/commit/0e33df147b4bf7e5d980a325953fddd73975cf32

The same pattern could be seen in the https://github.com/dotnet/msbuild/commit/2f3d37672a69142a13a62856b09034a915bedc70 that those changes are being reverted in this PR. 

Basically as part of the PR it is expected only files that are related to the scope of the problem to be changed.
Sorry for the confusions, it is important for us to have as granular changes as possible in order to identify possible regressions or bugs earlier:) 

Please correct me if there are mistakes in my steps of verification, I will be glad to double check it

Reply:
Are you asking that PRs not be kept in sync with the target branch? I haven't merged from vs17.9 which you requested earlier. Are you now requesting that not be done?

Reply:
We're in a weird state right now where changes are flowing into 17.9 and main, so reverting a merge from main is still resulting in an overall change of some unrelated things in this PR for 17.9. I think the best things to do would be to interactive-rebase onto 17.9. I can do that for you if you'd like.

Reply:
If it is not a problem, please go ahead and rebase the PR. (I can't give the PR much of my attention at present.) Thanks

Reply:
Not a problem, done!

Reply:
Resolving the conversation. Thank you @rainersigwald for the updates